From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Thu, 11 Jun 2020 12:58:55 +0000
Subject: cherry-pick: Don't remove DB account "domjudge@%" during uninstall

Since commit 6d84573c4 (No longer grant access to from the world to
the MySQL account, 2014-03-02), the only purpose of the removal is to
clean up after pre-4.0 versions of DOMjudge, and they are long gone.

The removal can cause issues, see issue 826.

Fixes issue 826.

(cherry picked from commit db311f0f93a2eaf61944cf386e08ebc05c3042b8)
---
 sql/dj_setup_database.in | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/sql/dj_setup_database.in b/sql/dj_setup_database.in
index 74ad35f12..584d2dbf7 100755
--- a/sql/dj_setup_database.in
+++ b/sql/dj_setup_database.in
@@ -209,10 +209,7 @@ remove_db_users()
 {
 	(
 	echo "DROP DATABASE IF EXISTS \`$DBNAME\`;"
-# Also drop "@%" users, since those may have been created by pre-4.0 versions
-# of DOMjudge.
-	echo "DROP USER IF EXISTS
-		'$domjudge_DBUSER'@'%', '$domjudge_DBUSER'@'localhost';"
+	echo "DROP USER IF EXISTS '$domjudge_DBUSER'@'localhost';"
 	echo "FLUSH PRIVILEGES;"
 	) | mysql -f
 	verbose "DOMjudge database and user(s) removed."
-- 
2.20.1

